.github/workflows/python-CI.yml

```diff
@@ -141,3 +141,26 @@ jobs:
         if: runner.os == 'Windows'
         run: |
           hatch run test:tests
+
+  test-launch-app:
+    runs-on: ${{ matrix.os }}
+    needs: changes
+    if: ${{ needs.changes.outputs.phoenix == 'true' }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, windows-2019, macos-12 ]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+      - uses: yezz123/setup-uv@v4
+        with:
+          uv-venv: test-launch-app
+      - run: uv pip install . opentelemetry-sdk portpicker psutil
+      - run: python -Xutf8 .github\.scripts\test_launch_app.py
+        if: runner.os == 'Windows'
+        timeout-minutes: 5
+      - run: python .github/.scripts/test_launch_app.py
+        if: runner.os != 'Windows'
+        timeout-minutes: 5
```

Comment:
non-blocking: Do you think it's worth doing this in a different workflow just so the erroring is cleaner? This runs AFTER all the above and it might just make the workflow slower and maybe this can be optimized independantly.

Reply:
yea, we can move it out, but it's currently running in parallel (only after the file change detection), so it already fails pretty fast (for the ubuntu version)